common/server.cpp

```diff
@@ -84,16 +84,28 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
         name = name.left(35);
 
     Server_DatabaseInterface *databaseInterface = getDatabaseInterface();
-
+    
     AuthenticationResult authState = databaseInterface->checkUserPassword(session, name, password, clientid, reasonStr, secondsLeft);
     if (authState == NotLoggedIn || authState == UserIsBanned || authState == UsernameInvalid || authState == UserIsInactive)
         return authState;
 
     ServerInfo_User data = databaseInterface->getUserData(name, true);
     data.set_address(session->getAddress().toStdString());
     name = QString::fromStdString(data.name()); // Compensate for case indifference
-
+    
     if (authState == PasswordRight) {
+        //check to see if user is full but allow users that donate access to server regardless of user count
+        if (!databaseInterface->isUserADonator(name)) {
```

Comment:
Im not so sure this is the correct solution.  The information being queried is then action-ed on.  The action to take shouldn't be what is queried from the DB.  There might be other things the information could be used for else were in the code (not really sure just a thought). Now we could say something like isUserAVIP as suggested by @skwerlman if the word "Donator" seems odd to everyone.

The general concept of the functionality is to mark a user as "important" but not "important enough" to create a new user role.  

@Daenyth 
Thoughts?